### PR TITLE
Mission - Streamline and add on-screen logging

### PR DESCRIPTION
**Pull request by Mike-MF**
- Will output as CBA Notify as: " [TAC Contamination Gas]: Marker size 62 larger than recommended size (60x60)."
- CBA Notify stacks which is nice.
- Dumps to RPT automatically, if optional arg is used will dump to RPT if on the server.
- Has some small minor formatting and changes to functions

> [TAC Reinforcements]: Too close to group: Alpha 1-6, on: WEST
> [TAC Earthquake]: Magnitude (6) not between 1-5, using default values.
> [TAC Earthquake]: Only one instance of Earthquake will happen locally.
> [TAC Reaction]: Unknown type value: stutic

Also can we force the CBA Notification lifetime to 4s with this? I had mine down to minimum because of ACRE notifications and I wouldn't have even seen them.

Minor Changes:
- Contamination Gas: Selects the greater value from marker size instead of the first.
- Contamination Gas: Doesn't give you a notification if the marker size is 60. Only greater than.
- Earthquake: Updated header.

### CODE REVIEW

File: addons/mission/functions/fnc_log.sqf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * Author: Mike
+ * Function based logging for 3DEN / Server
+ * Call from Functions
+ *
+ * Shows in-game warning and logs to RPT, optional logging to server/client RPT
+ *
+ * Arguments:
+ * 0: Function Name <STRING>
+ * 1: Error Message <STRING>
+ * 2: Log to Server <BOOL> (default: false)
```
Comment (jonpas) [inline addons/mission/functions/fnc_log.sqf:12]:
```suggestion
 * 2: Log to RPT <BOOL> (default: false)
```
What does this have to do with server specifically?

File: addons/mission/functions/fnc_log.sqf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Return Value:
+ * None
+ *
+ * Examples:
+ * ["[TAC Mission X]:", format ["Undefined value %1", _value]] call FUNC(log)
```
Comment (jonpas) [inline addons/mission/functions/fnc_log.sqf:18]: This is too verbose.
- Name should just be `Mission X`, because they are all `TAC` and that can be added in this function.
- Name should just use `__FILE__` preprocessor command.

This function should just be coupled with a macro to make the logging call as short as possible, eg. `MLOG("Undefined value %1",_value);` (and `MLOG_RPT` similar).

File: addons/mission/functions/fnc_log.sqf [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+params ["_functionName", "_message", ["_logToRPT", false]];
+
+private _message = [_functionName, _message] joinString " ";
+
+if (is3DENPreview) then {
+    _message call CBA_fnc_notify;
+    diag_log _message;
+} else {
+    if (_logToRPT) then {
```
Comment (jonpas) [inline addons/mission/functions/fnc_log.sqf:29]: Why have this at all? If it's a live mission, we won't see any warnings then, that's bad!

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/mission/functions/fnc_log.sqf (hunk 3 above)
Comment (Mike-MF) [reply addons/mission/functions/fnc_log.sqf:29]: Because on a live mission, there should be no errors. Mission reviews should catch them, and if they don't we'll see in server RPT

Comment (jonpas) [reply addons/mission/functions/fnc_log.sqf:29]: But we won't see in server RPT because:
- Of this condition...
- Because not every function runs on the server.

Reviews will not catch all, and I want to be able to yell at people who didn't check their missions properly.

Comment (Mike-MF) [reply addons/mission/functions/fnc_log.sqf:29]: The condition is false for local functions, Reinforcements and Reaction would both log to server RPT. (Or everyones if someone lets it run globally)

Comment (jonpas) [reply addons/mission/functions/fnc_log.sqf:29]: And again, why do we need that condition at all? It should always log to RPT.

**Review by jonpas — APPROVED** [review]
